compiler/AST/AggregateType.cpp

```diff
@@ -1782,8 +1782,9 @@ void AggregateType::buildConstructor() {
 }
 
 void AggregateType::buildDefaultInitializer() {
-  if (defaultInitializer                       == NULL ||
-      strcmp(defaultInitializer->name, "init") !=    0) {
+  if ((defaultInitializer                       == NULL ||
+      strcmp(defaultInitializer->name, "init") !=    0) &&
+      symbol->hasFlag(FLAG_REF) == false) {
```

Comment:
I'm confused as to why this case was necessary, since the ref class is now marked a "use default init".

Reply:
The 'use default init' prevents creation of a constructor for _ref, and this part disables creation of an initializer for _ref. The _ref type is so magic/special that I don't think we should try and treat it like a normal type. It's really a primitive type we happen to describe as a "class", for now.